src/TriggerBinding/SqlTriggerTargetScaler.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql
+{
+    public class SqlTriggerTargetScaler : ITargetScaler
+    {
+        private readonly string _userFunctionId;
+        private readonly ILogger _logger;
+        private readonly SqlTriggerMetricsProvider _metricsProvider;
+        private readonly int _maxChangesPerWorker;
+
+        public SqlTriggerTargetScaler(string userFunctionId, string userTableName, string connectionString, int maxChangesPerWorker, ILogger logger)
+        {
+            _ = !string.IsNullOrEmpty(userTableName) ? true : throw new ArgumentNullException(userTableName);
+            this._userFunctionId = !string.IsNullOrEmpty(userFunctionId) ? userFunctionId : throw new ArgumentNullException(userFunctionId);
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this._metricsProvider = new SqlTriggerMetricsProvider(connectionString, logger, new SqlObject(userTableName), userFunctionId);
+            this.TargetScalerDescriptor = new TargetScalerDescriptor(userFunctionId);
+            this._maxChangesPerWorker = maxChangesPerWorker;
+        }
+
+        public TargetScalerDescriptor TargetScalerDescriptor { get; }
+
+        public async Task<TargetScalerResult> GetScaleResultAsync(TargetScalerContext context)
+        {
+            SqlTriggerMetrics metrics = await this._metricsProvider.GetMetricsAsync();
+
+            return this.GetScaleResultInternal(context, metrics.UnprocessedChangeCount);
+        }
+
+        internal TargetScalerResult GetScaleResultInternal(TargetScalerContext context, long unprocessedChangeCount)
+        {
+            int concurrency = context.InstanceConcurrency ?? this._maxChangesPerWorker;
```

Comment:
What does "instance concurrency" mean and how is it related to the max changes per worker setting? Those two seem like slightly different concepts to me.

Reply:
InstanceConcurrency is a value provided by functions host when [dynamic concurrency](https://learn.microsoft.com/en-us/azure/azure-functions/functions-concurrency) is enabled, more information is provided in the link. The max changes per worker setting would correspond to a static concurrency setting.

Reply:
I did add this in the latest commit, with the other changes, do you want me to provide the link to dynamic concurrency and make the comment more detailed?